packages/core/strapi/ee/index.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+const { pick } = require('lodash/fp');
+
+const { readLicense, verifyLicense, fetchLicense, LicenseCheckError } = require('./license');
+const { eeStoreModel } = require('./ee-store');
+const { getRecurringCronExpression } = require('../lib/utils/cron');
+
+const ONE_MINUTE = 1000 * 60;
+const DEFAULT_FEATURES = {
+  bronze: [],
+  silver: [],
+  gold: ['sso'],
+};
+
+const ee = {
+  enabled: false,
+  licenseInfo: {},
+};
+
+const disable = (message) => {
+  ee.logger?.warn(`${message} Switching to CE.`);
+  // Only keep the license key for potential re-enabling during a later check
+  ee.licenseInfo = pick('licenseKey', ee.licenseInfo);
+  ee.enabled = false;
+};
+
+let initialized = false;
+
+const init = (licenseDir, logger) => {
+  // Can only be executed once, to prevent any abuse of the optimistic behavior
+  if (initialized) {
+    return;
+  }
+
+  initialized = true;
+  ee.logger = logger;
+
+  if (process.env.STRAPI_DISABLE_EE?.toLowerCase() === 'true') {
+    return;
+  }
+
+  try {
+    const license = process.env.STRAPI_LICENSE || readLicense(licenseDir);
+
+    if (license) {
+      ee.licenseInfo = verifyLicense(license);
+      ee.enabled = true; // Optimistically enable EE during initialization
+    }
+  } catch (error) {
+    disable(error.message);
+  }
+};
+
+const onlineUpdate = async ({ strapi }) => {
+  const transaction = await strapi.db.transaction();
+
+  try {
+    // TODO: Use the core store interface instead, it does not support transactions and "FOR UPDATE" at the moment
```

Comment:
it should support transactions soon enough

Reply:
Actually, this comment is outdated. I'm gonna remove it. We're not using the core store anymore, but maybe a similar interface could be created for the ee_store? Not sure it's warranted but why not